src/modules/expression-evaluator/src/evaluateExpressionTestData.ts

```diff
@@ -527,6 +527,69 @@ testData.singleApplicationProperty_depth2 = {
   children: ['application.questions.q2'],
 }
 
+// String substitution
+
+testData.stringSubstitutionSingle = {
+  operator: 'stringSubstitution',
+  children: ['Hello, %1, welcome to our site.', 'friend'],
+}
+
+testData.stringSubstitutionMultiple = {
+  operator: 'stringSubstitution',
+  children: [
+    'There are %1 kinds of people in the world:\nthose who understand %2 and those who %3',
+    '10',
+    { value: 'binary' },
+    "don't",
+  ],
+}
+
+testData.stringSubstitutionNonStringReplacements = {
+  operator: 'stringSubstitution',
+  children: ['We have %1 people listed: %2', 2, { value: ['Boba', 'Mando'] }],
+}
+
+testData.stringSubstitutionTooManyReplacements = {
+  operator: 'stringSubstitution',
+  children: ['The price of milk is %1 per %2', '$2.30', 'liter', 'gallon', '$5.00'],
```

Comment:
Should this be using `number` instead of string?

Reply:
Use `number` where? 😕

Reply:
Okay, I've updated the earlier "non-string replacements" test to include a float number as well as an integer:
```
{
  operator: 'stringSubstitution',
  children: [
    'We have %1 %2 listed with an average value of %3: %4',
    2,
    'people',
    4.53,
    { value: ['Boba', 'Mando'] },
  ],
}
```